app/src/main/java/com/zulip/android/filters/NarrowFilterByDate.java

```diff
@@ -3,23 +3,25 @@
 import android.os.Parcel;
 import android.os.Parcelable;
 
-import com.j256.ormlite.stmt.SelectArg;
 import com.j256.ormlite.stmt.Where;
 import com.zulip.android.models.Message;
 import com.zulip.android.models.Stream;
+import com.zulip.android.util.Constants;
 
 import org.json.JSONException;
 
 import java.sql.SQLException;
 import java.text.SimpleDateFormat;
+import java.util.Calendar;
 import java.util.Date;
+import java.util.Locale;
 
 public class NarrowFilterByDate implements NarrowFilter {
 
     public static final Parcelable.Creator<NarrowFilterByDate> CREATOR = new Parcelable.Creator<NarrowFilterByDate>() {
         public NarrowFilterByDate createFromParcel(Parcel in) {
-
-            return new NarrowFilterByDate();
+            in.setDataPosition(0);
+            return new NarrowFilterByDate((Date) in.readValue(Date.class.getClassLoader()));
```

Comment:
Awesome :) Only one comment though, we don't need to set data position to initial since its already set in the beginning when createFromParcel is called :)